crates/bevy_ecs/src/archetype.rs

```diff
@@ -296,7 +298,7 @@ pub struct Archetype {
     table_id: TableId,
     edges: Edges,
     entities: Vec<ArchetypeEntity>,
-    components: ImmutableSparseSet<ComponentId, ArchetypeComponentInfo>,
+    components: HashMap<ComponentId, ArchetypeComponentInfo>,
```

Comment:
Just a off-hand comment. But @james7132 was researching how we could generate perfect hash functions (PHF). With a PHF we could be using a dense array here. Especially if this field `components` is still immutable.

Reply:
Anecdotally I investigated a PHF crate for a similar purpose in a project of my own and found it to be much slower than FxHash, so I'm somewhat skeptical of how performant it might be; but that may have just been the crate being poorly optimized, or some other factor.